ts/screens/preferences/DefaultCalendarSelectScreen.tsx

```diff
@@ -0,0 +1,196 @@
+import * as pot from "italia-ts-commons/lib/pot";
+import * as React from "react";
+import { connect } from "react-redux";
+
+import { NavigationInjectedProps } from "react-navigation";
+
+import {
+  Platform,
+  StyleSheet,
+  TouchableHighlight,
+  TouchableOpacity,
+  View
+} from "react-native";
+
+import { Button, Text } from "native-base";
+
+import I18n from "../../i18n";
+
+import { ReduxProps } from "../../store/actions/types";
+import { GlobalState } from "../../store/reducers/types";
+
+import TopScreenComponent from "../../components/screens/TopScreenComponent";
+import Markdown from "../../components/ui/Markdown";
+import { preferredCalendarSaveSuccess } from "../../store/actions/persistedPreferences";
+
+import RNCalendarEvents, { Calendar } from "react-native-calendar-events";
+import customVariables from '../../theme/variables';
+import IconFont from "../../components/ui/IconFont";                                                
+
+type OwnProps = NavigationInjectedProps;
+
+type Props = ReturnType<typeof mapStateToProps> &
+  ReduxProps &
+  OwnProps;
+
+const styles = StyleSheet.create({
+  calendarItemWrapper: {
+    flexDirection: "row",
+    justifyContent: "space-between",
+    paddingVertical: 16,
+    borderBottomWidth: 1,
+    borderBottomColor: customVariables.brandLightGray,
+    color: customVariables.contentPrimaryBackground
+  }
+});
+
+const TouchableComponent =
+  Platform.OS === "ios" ? TouchableHighlight : TouchableOpacity;
+
+type CalendarItemProps = {
+  calendar: Calendar;
+  onPress: ()=> void;
+};
+
+/**
+ * Renders a Calendar as FlatList item
+ */
+const CalendarItem: React.SFC<CalendarItemProps> = props => (
+  <TouchableComponent
+    onPress={props.onPress}
+    style={styles.calendarItemWrapper}
+  >
+    <Text link={true}>{props.calendar.title}</Text>
+    <IconFont
+      name="io-right"
+      color={customVariables.contentPrimaryBackground}
+    />
+  </TouchableComponent>
+);
+  
+  
+type State = {
+  calendars: pot.Pot<ReadonlyArray<Calendar>, ResourceError>;
+};
+
+type FetchError = {
```

Comment:
```suggestion
type FetchError = "FETCH_ERROR";
```

Reply:
`kind` is used for types that have attributes

Reply:
ditto

Reply:
ditto